src/inc/corprof.idl

```diff
@@ -2375,6 +2375,34 @@ interface ICorProfilerCallback7 : ICorProfilerCallback6
 }
 
 
+[
+    object,
+    uuid(5BED9B15-C079-4D47-BFE2-215A140C07E0),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerCallback8 : ICorProfilerCallback7
+{
+    // This event is triggered whenever a metadata less method is jit compiled.
+    // These include IL Stubs, DynamicMethod or more generally any use of LCG.
+    // The goal is to provide profiler writers with enough information to identify
+    // it to users as beyond unknown code addresses.
+    // Note: FunctionID's provided here cannot be used to resolve to their metadata
+    //       tokens since LCG methods have no metadata.
+    HRESULT LightweightJITCompilationStarted(
+        FunctionID functionId,
+        UINT_PTR   ilHeader,
+        [in, string] const WCHAR * className,
+        [in, string] const WCHAR * methodName,
+        [in, string] const WCHAR * methodSignature);
+
```

Comment:
@noahfalk What are your thoughts on fixing GetFunctionFromIP to return metadata-less functions? I'll also look at seeing what the interaction is for GetCodeInfo and GetILToNativeMapping just for completeness.

I can instead make a GetNoMetadataFunctionFromIP, that the author will have to call if he cares about it.

I'm leaning towards GetNoMetadataFunctionFromIP since it'll not modify the behavior of our APIs, but then we look at these APIs a few years from now and we'll look at the clumsy things that software versioning causes us to do :)